scanpointgenerator/generators/lissajousgenerator.py

```diff
@@ -54,25 +59,27 @@ def __init__(self, names, units, box, num_lobes, num_points=None):
 
         self.axes = self.names  # For GDA
 
-    def _calc(self, i):
-        """Calculate the coordinate for a given index"""
-        x = self.centre[0] + \
-            self.x_max * m.sin(self.x_freq * i * self.increment +
-                               self.phase_diff)
-        y = self.centre[1] + \
-            self.y_max * m.sin(self.y_freq * i * self.increment)
-
-        return x, y
-
-    def iterator(self):
-        for i in range_(self.num):
-            p = Point()
-            p.positions[self.names[0]], p.positions[self.names[1]] = self._calc(i)
-            p.lower[self.names[0]], p.lower[self.names[1]] = self._calc(i - 0.5)
-            p.upper[self.names[0]], p.upper[self.names[1]] = self._calc(i + 0.5)
-            p.indexes = [i]
-
-            yield p
+    def produce_points(self):
+        self.points = {}
+        self.bounds = {}
+
+        x0, y0 = self.centre[0], self.centre[1]
+        A, B = self.x_max, self.y_max
+        a, b = self.x_freq, self.y_freq
+        d = self.phase_diff
+        f = lambda t: x0 + A * np.sin(a * 2 * m.pi * t/self.num + d)
```

Comment:
I'd like to keep the x and y axes formulae distinct just so visual recognition of the normal parametric system is obvious, but merging the normal and bounds case is a good idea. The difference should be captured in the passed 't', not the lambda.